clang/runtime/dpct-rt/include/dpct/math.hpp

```diff
@@ -2047,6 +2047,70 @@ class joint_matrix {
   matrix_accessor x;
   const size_t num_elements;
 };
+
+template <typename T>
+void ldmatrix(uintptr_t addr, T *m, const sycl::nd_item<3> &item_ct1,
```

Comment:
The condition for this helper function to work correctly is that the size of subgroup for the kernel function is 32, shown in https://github.com/oneapi-src/SYCLomatic/blob/b2a64459371c66a40896ce0f0a52189fb0b99d03/clang/test/dpct/cub/warplevel/warpscan.cu#L518
You can grep "SUBGROUPSIZE_FACTORY" and  "addSubGroupSizeRequest" to see how to add this attribute.

Reply:
Done

Reply:
I have not found the implementation logic for this part . Please add the lit test for this part.

Reply:
Lit test is added [here](https://github.com/oneapi-src/SYCLomatic/pull/2692/files#diff-d7f820548b14403f60cbeda950e4ea257b2cd5f7205c6d0f79908e05ccb3fbb4R94)